base/special/trig.jl

```diff
@@ -158,7 +159,7 @@ function asin{T<:AbstractFloat}(x::T)
         if absx == one(T)
             return flipsign(T(pi)/2, x)
         end
-        throw(DomainError(x, "asin(x) is not defined for |x|>1."))
+        noinline_domainerror(x, "asin(x) is not defined for |x|>1.")
```

Comment:
is this style or performance? I'm going to do the same in other functions and it seems a bit odd to have one for `acos` as well, but if it's needed for performance, I'll do it ofc :)